### PR TITLE
kucoinfutures fundingTimestamp

### PR DESCRIPTION
**Pull request by leastchaos**
This is update from https://github.com/ccxt/ccxt/pull/14393 as it was closed.
The PR was closed by @samgermain due to the concern that the funding rate interval may change on a random day and may not be fixed. 
1. I think granularity is provided by the kucoin API. If the funding rate interval changes, the granularity response would also be updated to reflect the current interval. So it should not be impacted? 
2. https://futures.kucoin.com/contract/history-rate the funding rate history shows it's every 8 hours for a long time for all markets so its unlikely they will randomly change.
It would be useful to be able to directly get the fundingTimestamp as the info is already available here.

### CODE REVIEW

**Comment by leastchaos** [comment]
could this be reviewed?

**Comment by samgermain** [comment]
@frosty00 It looks good to me, what do you think

File: js/kucoinfutures.js [hunk 1]
```diff
@@ -1504,8 +1504,10 @@ module.exports = class kucoinfutures extends kucoin {
         //    }
         //
         const data = this.safeValue (response, 'data');
-        const fundingTimestamp = this.safeNumber (data, 'timePoint');
-        // the website displayes the previous funding rate as "funding rate"
+        const previousFundingTimestamp = this.safeInteger (data, 'timePoint');
+        const granularity = this.safeInteger (data, 'granularity');
+        const fundingTimestamp = this.sum (previousFundingTimestamp, granularity);
+        // the website displays the previous funding rate as "funding rate"
```
Comment (samgermain) [inline js/kucoinfutures.js:1510]: could you move this comment up to under `const previousFundingTimestamp = this.safeInteger (data, 'timePoint');`

**Review by samgermain — CHANGES_REQUESTED** [review]

**Review by samgermain — APPROVED** [review]

**Comment by carlosmiei** [comment]
[🤖 TestsManager 🤖  Id: 63ec0cfa3631fae457232518]
 ```

Testing { exchanges: ["kucoinfutures"], symbol: "all", keys: { '--js': true, '--php': false, '--python': false, '--python-async': true, '--php-async': false, '--privateOnly': true }, maxConcurrency: 5 } (run-tests.js:255)
[100%] Testing kucoinfutures [AuthenticationError] [ExchangeNotAvailable] [ExchangeNotAvailable] (testExchange @ run-tests.js:172)

WARN kucoinfutures (JavaScript): (explain @ run-tests.js:191)

TESTING { exchange: 'kucoinfutures', symbol: 'all' } 

138 symbols []
SYMBOL: BTC/USDT:USDT
Testing kucoinfutures loadMarkets ( )
Testing kucoinfutures fetchCurrencies ( )
Testing kucoinfutures fetchTicker ( BTC/USDT:USDT )
2023-02-14T22:37:08.134Z kucoinfutures fetchTicker BTC/USDT:USDT 22178
BTC/USDT:USDT fetchTicker 2023-02-14T22:37:08.134Z bid: 22178 ask: 22179
Testing kucoinfutures fetchTickers ( BTC/USDT:USDT )
Testing kucoinfutures fetchOHLCV ( BTC/USDT:USDT )
BTC/USDT:USDT fetched 10 OHLCVs
Testing kucoinfutures fetchTrades ( BTC/USDT:USDT )
BTC/USDT:USDT fetched 100 trades
Testing kucoinfutures fetchOrderBook ( BTC/USDT:USDT )
BTC/USDT:USDT fetchOrderBook 1672837631553 20 bids: [ 22178, 8591 ] 20 asks: [ 22179, 25160 ]
Testing kucoinfutures fetchL2OrderBook ( BTC/USDT:USDT )
BTC/USDT:USDT fetchL2OrderBook 1672837631556 20 bids: [ 22178, 8591 ] 20 asks: [ 22179, 25160 ]
Testing kucoinfutures fetchOrderBooks ( )
Testing kucoinfutures fetchBalance ( )
fetching balance...
[AuthenticationError] kucoinfutures KC-API-KEY not exists (explain @ run-tests.js:193)

WARN kucoinfutures (Python 3 Async): (explain @ run-tests.js:191)

[ExchangeNotAvailable] kucoinfutures GET https://cors-anywhere.herokuapp.com/https://api-futures.kucoin.com/api/v1/ticker?symbol=XBTUSDTM 403 Forbidden See /corsdemo for more info
kucoinfutures using proxy ``
EXCHANGE: kucoinfutures
SYMBOL: BTC/USDT:USDT
CODE: BTC
Testing fetch_ticker:BTC/USDT:USDT
kucoinfutures BTC/USDT:USDT ticker 2023-02-14T22:37:16.500Z high: None low: None bid: 22178.0 ask: 22179.0 volume: None
Testing fetch_tickers:BTC/USDT:USDT
Testing fetch_ohlcv:BTC/USDT:USDT
kucoinfutures fetched 10 OHLCVs
Testing fetch_order_book:BTC/USDT:USDT
kucoinfutures BTC/USDT:USDT order book 2023-02-14T22:37:17.241Z bid: 22178.0 bidVolume: 10631.0 ask: 22179.0 askVolume: 25903.0
Testing fetch_trades:BTC/USDT:USDT
kucoinfutures BTC/USDT:USDT fetched 100 trades
Testing fetch_orders:BTC/USDT:USDT
kucoinfutures BTC/USDT:USDT fetchOrders() is not supported
Testing fetch_open_orders:BTC/USDT:USDT
{'type': 'AuthenticationError', 'num_retries': 0, 'max_retries': 2} kucoinfutures KC-API-KEY not exists
kucoinfutures using proxy `https://cors-anywhere.herokuapp.com/`
EXCHANGE: kucoinfutures
SYMBOL: BTC/USDT:USDT
CODE: BTC
Testing fetch_ticker:BTC/USDT:USDT
{'type': 'ExchangeNotAvailable', 'num_retries': 1, 'max_retries': 2} kucoinfutures GET https://cors-anywhere.herokuapp.com/https://api-futures.kucoin.com/api/v1/ticker?symbol=XBTUSDTM 403 Forbidden See /corsdemo for more info
[ExchangeNotAvailable] kucoinfutures GET https://cors-anywhere.herokuapp.com/https://api-futures.kucoin.com/api/v1/ticker?symbol=XBTUSDTM 403 Forbidden See /corsdemo for more info (explain @ run-tests.js:193)
WARN ["kucoinfutures"] (run-tests.js:272)
All done, 1 warnings (run-tests.js:276)


```

**Comment by samgermain** [comment]
reopened in #18867

**Pull request closed without merging**